submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -0,0 +1,123 @@
+const borders = {
+    rightBorder: 400,
+    leftBorder: 5,
+    upBorder: 410,
+    downBorder: 5
+};
+const someGap = 10;
+
+class GameSprite {
+    constructor (x, y,sprite)  {
+        this.sprite = sprite;
+        this.x = x;
+        this.y = y;
+    }
+    render = function () {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+}
+
+class Enemy extends GameSprite {
+    constructor(x, y, sprite = "images/enemy-bug.png", speed = "75") {
+        super(x, y, sprite);
+        this.x = x;
+        this.y = y;
+        this.speed = speed;
+        this.sprite = sprite;
+    }
+    render(){};
+};
+
+    Enemy.prototype.update = function (dt) {
+        this.x += this.speed * dt;
+        const addSpeed = 40;
+        if (this.x >= 500) {
+        this.x = -someGap;
+        this.speed = Math.floor(addSpeed + Math.random() * 100);
+        }
+    };
+
+this.checkCollisions = function () {
```

Comment:
this is global definition of checkCollisions function, because 'this' in this case is just 'window'...
maybe it's better to define it with 'var' or 'const', or just like a function...

Reply:
made a function. Thanks.